specs/Deployment/MSIXPackageVersioning.md

```diff
@@ -0,0 +1,517 @@
+
+* Major = 0
+* Minor = 5
+* Year = 21 (aka 2021)
+* Month = 03
+* Day = 25
+* Patch = 000 (if we have a 0.5.1 release this would be `001`)
+
+This doesn't scale going forward e.g. MSIX version is a DotQuadNumber aka `uint16[4]`. If 0.8 was released on May'31, 2021 MSIX version would be `0.82105.31000.0`
+
+### 2.4.2. WinUI 2.x Notation
+
+WinUI 2.x encodes MSIX versions as `M.NYYMM.DDPPP.0` (same as Windows App SDK 0.5). This runs into issues starting with WinUI 2.7, as identified by [WinUI version schema sometime can't applied to MinVersion #4008](https://github.com/microsoft/microsoft-ui-xaml/issues/4008#issuecomment-765633788).
+
+## 2.5. Decision 5: Package Names
+
+Windows App SDK 0.8+ has 3 MSIX packages. We have choices for package Name patterns for 0.8 and 1.x.
+
+Assuming Breaking Change Boundary for v0.x and v1+'s decision are `Major.Minor` version we have the
+following naming patterns...
+
+* WARfwk: `Microsoft.WindowsAppRuntime.<rmajor>.<rminor>[-tag]`
+* WARmain: `Microsoft.WindowsAppRuntime.Main.<rmajor>.<rminor>[-tag]`
+* WARddlm: `Microsoft.WindowsAppRuntime.DDLM.<major>.<minor>.<build>.<revision>-<shortarchitecture>[-shorttag]`
+
+where
+
+* rmajor = Major version number of the project release, base-10, no leading zeros (e.g. "1" for WindowsAppSDK 1.2)
+* rminor  = Minor version number of the project release, base-10, no leading zeros (e.g. "2" for WindowsAppSDK 1.2)
+* major = Major version number of the release, base-10, no leading zeros
+* minor  = Minor version number of the release, base-10, no leading zeros
+* build = Build version number, base-10, no leading zeros
+* revision = Revision version number, base-10, no leading zeros
+* architecture = Allowed values: "x86", "x64", "arm64"
+* shortarchitecture = Allowed values: "x8", "x6", "a6"
+* tag = Allowed values: "", "preview[#]", "experimental[#]"
+* shorttag = Allowed values: "", "p[#]", "e[#]"
+
+**NOTE:** rmajor/rminor are the release version, major/minor/build/revision are the MSIX package
+version (Microsoft.ProjectReunion.0.8-preview had a release version of 0.8
+but an MSIX package version of 8000.146.628.0).
+
+Version's fields have values 0-65535.
+
+WARddlm requires architecture in its package Name to support concurrent use of different
+architectures of a framework. See the [Dynamic Dependencies spec](https://github.com/microsoft/WindowsAppSDK/blob/main/specs/dynamicdependencies/DynamicDependencies.md)
+for more details.
+
+This leads to package Name length issues even for common cases:
+
+|Package|Average|AverageLength|
+| --- | :--- | :---: |
+|WARfwk |Microsoft.WindowsAppRuntime.1.15-preview1|41|
+|WARmain|Microsoft.WindowsAppRuntime.Main.1.15-preview1|46|
+|WARmain|Microsoft.WindowsAppRuntime.Singleton-preview1|46|
+|WARddlm|Microsoft.WindowsAppRuntime.DDLM.1.15.12345.24680-arm64-preview1|**<span style="color:red">64</span>**|
+
+|Package|Min|MinLength|
+| --- | :--- | :---: |
+|WARfwk |Microsoft.WindowsAppRuntime.1.0-preview1|40|
```

Comment:
Should we use `WindowsAppRT` (even though it has an unfortunate acronym and association with previous OS efforts) here?

Reply:
No. We can make the naming work, no need to raise shades of times best forgot 😋